internal/cmd/commands/dev/dev.go

```diff
@@ -718,6 +718,10 @@ func (c *Command) Run(args []string) int {
 			c.Info["worker auth current key id"] = c.worker.WorkerAuthCurrentKeyId
 			c.InfoKeys = append(c.InfoKeys, "worker auth storage path")
 			c.Info["worker auth storage path"] = c.worker.WorkerAuthStorage.BaseDir()
+			if err := c.StoreWorkerAuthReq(c.worker.WorkerAuthRegistrationRequest, c.worker.WorkerAuthStorage.BaseDir()); err != nil {
+				c.UI.Error(fmt.Errorf("Error storing workerAuth request: %w", err).Error())
+				return base.CommandUserError
```

Comment:
This gets a little ugly. See above how if Start fails we shut down the worker? If we're going to exit here we should really do a proper shutdown of the worker and controller.

I was going to suggest doing something like `errorEncountered.Store(true)` as in the goroutine but in fact that logic is already faulty. If there _is_ an error we skip out of the for loop and return command success 😬 If you're up for refactoring it, I think what needs to happen is that you could output the error on the UI (as in the goroutine below) and set encountered error here to true and continue; that will cause the `for` loop below not to run; and everything in the for loop in the `if shutdownTriggered` bit needs to go after the `for` loop, and then whether success or error is returned as the return code depends on if errorEncountered was true or not.